### PR TITLE
only show the repo thing and the navbar when there is a repo path

### PR DESCRIPTION
**Pull request by colinwerner**
There were a couple ways to go about doing this, this was the simpliest.

Basically, only show the repo thing and the navbar when there is a repo path.

### CODE REVIEW

File: app/misc/router.ts [hunk 1]
```diff
@@ -311,6 +314,22 @@ function hideDiffPanelButtons() {
...
+    document.getElementById("nav-repo-branch-tag-info")!.style.visibility = "hidden";
+    document.getElementById("nav-toolbar")!.style.visibility = "hidden";
+  }
+}
+
+function displayNavToolbar() {
+  // only show these if there is a repo
+  if (repoFullPath) {
+    document.getElementById("nav-repo-branch-tag-info")!.style.visibility = "visible";
+    document.getElementById("nav-toolbar")!.style.visibility = "visible";
```
Comment (roweba) [inline app/misc/router.ts]: Maybe you should try using the display property?

`document.getElementById("nav-repo-branch-tag-info")!.style.display = "none";` to hide
`document.getElementById("nav-repo-branch-tag-info")!.style.display = "block";` to show

**Comment by maxUser** [comment]
The back button allows the user to access the graph page without having a repo open. Maybe remove it until a repo has been opened?

image

File: app/misc/router.ts (hunk 1 above)
Comment (colinwerner) [reply app/misc/router.ts]: I don't think it's an issue with the method. I think it's an issue with the repoFullPath?

**Comment by colinwerner** [comment]
> The back button allows the user to access the graph page without having a repo open. Maybe remove it until a repo has been opened?
> 
> image

Good catch. I'm not sure the best way to see if there is a repo open or not, any ideas?

**Comment by colinwerner** [comment]
> Works on windows, the icon next to the repo navbar is still always visible but I think that's intentional and looks nice

Yep, that was intentional.

File: app/misc/router.ts [hunk 2]
```diff
@@ -311,6 +314,31 @@ function hideDiffPanelButtons() {
...
+      repoCreate.value || repoClone.value || repoLocalPath)
+  {
+    showRepoNavTools = "visible";
+  }
+
+  // show/hide the relevent items
+  document.getElementById("repo-back-button")!.style.visibility = showRepoNavTools;
+  document.getElementById("nav-repo-branch-tag-info")!.style.visibility = showRepoNavTools;
+  document.getElementById("nav-toolbar")!.style.visibility = showRepoNavTools;
+  document.getElementById("nav-open-repo-button")!.style.visibility = showRepoNavTools;
```
Comment (mehrler12) [inline app/misc/router.ts]: Why'd you hide the folder button? I liked it being always visible

Comment (colinwerner) [reply app/misc/router.ts]: You and @maxUser can duke it out... let me know the result.

Comment (hdauvic) [reply app/misc/router.ts]: I concur with @mehrler12 but in addition to this if I do not have a repo open, there's no clear way of returning to the "open a repo" screen without the folder button. And as of right now (I know there's an issue to fix this) but visualgit always opens to that screen so now I can't open a repo at all.

Comment (colinwerner) [reply app/misc/router.ts]: > I concur with @mehrler12 but in addition to this if I do not have a repo open, there's no clear way of returning to the "open a repo" screen without the folder button. And as of right now (I know there's an issue to fix this) but visualgit always opens to that screen so now I can't open a repo at all.
> This is the screen visualgit opens to, and I can do nothing now as I have no repo open and no way to open one?
> repofolder

We either have to abandon this whole change or live with it. Why does VG open to this screen is the question?

Comment (roweba) [reply app/misc/router.ts]: Tested it with the new updates and I'm having the same problem as @hdauvic. VisualGit opens to the graph page by default, so I have no way of getting to the "open repo" page. I think you should either add the folder back or find a way to open VisualGit to the correct page.

**Comment by colinwerner** [comment]
@maxUser you haven been outvoted ... for now.

@mehrler12 @roweba @hdauvic  please re-test now.

File: app/misc/router.ts (hunk 2 above)
Comment (colinwerner) [reply app/misc/router.ts]: "Please try again." :)

**Pull request merged**